packages/expo-cli/e2e/__tests__/start-test.ts

```diff
@@ -27,3 +28,34 @@ test('start --offline', async () => {
   });
   cli.stderr.pipe(process.stderr);
 });
+
+test('start --offline with existing packager info', async () => {
+  await ProjectSettings.setPackagerInfoAsync(tempDir, {
+    devToolsPort: 19002,
+    expoServerPort: 19001,
+    packagerPort: 19000,
+    packagerPid: 1337,
+  });
+
+  const promise = spawnAsync(EXPO_CLI, ['start', '--offline'], { cwd: projectDir });
+  const cli = promise.child;
+
+  cli.stderr.pipe(process.stderr);
+  cli.stdout.on('data', data => {
+    process.stdout.write(data);
+    if (/Your native app is running/.test(data.toString())) {
+      cli.kill('SIGINT');
+    }
+  });
+
+  try {
+    await promise;
```

Comment:
I had to wait until the command exits with `SIGINT`, else the test would just validate even though it never closes. Not sure if that's just my computer, or if this is also missing from the basic `start --offline` test.

Reply:
Talked this over with Ville, we should `await` else Jest will skip the test and not wait for the results.